pkg/operator/ceph/controller/spec.go

```diff
@@ -51,6 +54,32 @@ type daemonConfig struct {
 
 var logger = capnslog.NewPackageLogger("github.com/rook/rook", "ceph-spec")
 
+var (
+	cronLogRotate = `
+set -xe
+
+CEPH_CLIENT_ID=%s
+PERIODICITY=%s
+LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
+
+if [ -z "$PERIODICITY" ]; then
+	PERIODICITY=24h
+fi
+
+# edit the logrotate file to only rotate a specific daemon log
+# otherwise we will logrotate log files without reloading certain daemons
+# this might happen when multiple daemons run on the same machine
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+
+while true; do
```

Comment:
It seems like log rotation happens both (1) when pods start and (2) at the PERIODICITY interval. I think maybe the docs should mention this?

Reply:
I've decided not to rotate immediately once the container starts, we never do this outside of the container world. I don't see a good reason to do things differently in containers. So we now rotate after the periodicity flag.